tests/test_updater_ng.py

```diff
@@ -148,6 +149,35 @@ def test_refresh_with_only_local_root(self):
     # Get targetinfo for 'file3.txt' listed in the delegated role1
     targetinfo3= self.repository_updater.get_one_valid_targetinfo('file3.txt')
 
+
+  def test_is_target_in_pathpattern(self):
+    supported_use_cases = [
+      ("foo.tgz", "foo.tgz"),
+      ("foo.tgz", "*"),
+      ("foo.tgz", "*.tgz"),
+      ("foo-version-a.tgz", "foo-version-?.tgz"),
+      ("targets/foo.tgz", "targets/*.tgz"),
+      ("foo/bar/zoo/k.tgz", "foo/bar/zoo/*"),
+      ("foo/bar/zoo/k.tgz", "foo/*/zoo/*"),
+      ("foo/bar/zoo/k.tgz", "*/*/*/*"),
+      ("foo/bar", "f?o/bar"),
+    ]
```

Comment:
I think the test `("foo/bar", "*o/bar")` tests the same use case as line 162:
`("foo/bar/zoo/k.tgz", "*/*/*/*"),`.
Even in line 162 you make sure you use `fnmatch.fnmatch()` on all 4 directories otherwise, it cannot be successful.

Reply:
I'm just saying: your earlier PR passed all those tests but when I added `"*o/bar"` it did not pass... 

Yes I can see the current implementation should succeed but this is basic regression testing: if you fix a bug, add a test case so the bug won't come back.

Reply:
I just added that test.